tests/diff.test.js

```diff
@@ -0,0 +1,446 @@
+const diff = require('./diff');
+const jsdom = require('JSDom');
+
+function vnode(tag,children,props,css,ns,ref,oc,od,key) {
+  return { 'type': 'vnode'
+         , 'tag' : tag
+         , 'children' : children
+         , 'props' : props
+         , 'css' : css
+         , 'ns' : ns
+         , 'domRef' : ref
+         , 'onCreated' : oc
+         , 'onDestroyed' : od
+         , 'key' : key
+         };
+}
+
+function vtext(txt) {
+  return { 'type': 'vtext'
+         , 'text' : txt
+         };
+}
+
+// base case
+test('Should be null when diffing two null virtual DOMs', () => {
+  const document = new jsdom.JSDOM().window.document;
+  const body = document.body;
+  var c = null; n = null;
+  diff(c, n, body, document);
+  expect(c).toBe(null);
```

Comment:
I think it may be impossible for `c` *not* to be `null`. Regardless of how `diff` is implemented, it cannot alter the value of a variable in the scope from which it is called. With `c` and `n` being defined as `null`, there's nothing `diff` could do to change that.

Reply:
Yea, you're right, `c` and `n` cannot be altered in this scope if they're `null`. A better test here might be `expect(body.childNodes.length).toBe(0);`